src/devices/Ads1115/Ads1115.cs

```diff
@@ -104,14 +220,255 @@ private void SetConfig()
+        /// <param name="lowerValue">Lower value for the comparator</param>
+        /// <param name="upperValue">Upper value for the comparator</param>
+        /// <param name="mode">Traditional or Window comparator mode</param>
+        /// <param name="queueLength">Minimum number of samples that must exceed the threshold to trigger the event</param>
+        /// <param name="gpioController">The GPIO controller, to configure the pin</param>
+        /// <param name="pin">The pin where the ALRT/RDY pin is connected to. </param>
+        public void EnableComparator(short lowerValue, short upperValue, ComparatorMode mode,
+            ComparatorQueue queueLength, GpioController gpioController, int pin)
+        {
+            if (gpioController == null)
+            {
+                throw new ArgumentNullException(nameof(gpioController));
+            }
+
+            if (queueLength == ComparatorQueue.Disable)
+            {
+                throw new ArgumentException("Must set the ComparatorQueue to something other than disable.", nameof(queueLength));
+            }
+
+            if (upperValue <= lowerValue)
+            {
+                throw new ArgumentException("Lower comparator limit must be larger than upper comparator limit");
+            }
+
+            if (_gpioController != null)
+            {
+                throw new InvalidOperationException("Event mode already set.");
+            }
+
+            try
+            {
+                // The ALRT/RDY Pin requires a pull-up resistor
+                gpioController.OpenPin(pin, PinMode.InputPullUp);
+
+                _comparatorQueue = queueLength;
+                _comparatorMode = mode;
+                // Event callback mode is only useful in Continuous mode
+                _deviceMode = DeviceMode.Continuous;
+                SetConfig();
+                // Writing a negative value to the max value register and a positive value to the min value register
+                // configures the ALRT/RDY pin to trigger after each conversion (with a transition from high to low, when ComparatorPolarity is Low)
+                WriteComparatorRegisters(lowerValue, upperValue);
+
+                gpioController.RegisterCallbackForPinValueChangedEvent(pin, ComparatorPolarity == ComparatorPolarity.Low ? PinEventTypes.Falling : PinEventTypes.Rising, ConversionReadyCallback);
+
+            }
+            catch (Exception)
+            {
+                gpioController.ClosePin(pin);
+                throw;
+            }
+
+            _gpioController = gpioController;
+            _alrtRdyPin = pin;
+        }
+
+        private ushort ReadConfigRegister()
+        {
+            Span<byte> retBuf = stackalloc byte[2];
+            Span<byte> request = stackalloc byte[1]
+            {
+                (byte)Register.ADC_CONFIG_REG_ADDR
+            };
+
+            _i2cDevice.WriteRead(request, retBuf);
+            return BinaryPrimitives.ReadUInt16BigEndian(retBuf);
```

Comment:
presumably first byte of retBuf will always be zero (most of the devices would do that since you're writing register in the first byte) - if that's the case please just return byte here

Reply:
I don't quite understand what you mean, but this reads the 16 bit wide config register (register 0). And all bits are used and returned here.